lib/services/role.go

```diff
@@ -1557,52 +1557,6 @@ func (set RoleSet) CheckGCPServiceAccounts(ttl time.Duration, overrideTTL bool)
 	return utils.StringsSliceFromSet(accounts), nil
 }
 
-// CheckAccessToSAMLIdP checks access to the SAML IdP.
-//
-// TODO(sshah): Delete after enterprise changes is merged to use v2.
-//
-//nolint:revive // Because we want this to be IdP.
-func (set RoleSet) CheckAccessToSAMLIdP(authPref readonly.AuthPreference, state AccessState) error {
```

Comment:
Can we just rename `CheckAccessToSAMLIdPV2` to `CheckAccessToSAMLIdP` now? It seems strange to have V2 in the code when there's no V1 anymore.

Reply:
The `CheckAccessToSAMLIdP` still lives, its just unexported and with an updated signature but the internal access checking logic remains the same. https://github.com/gravitational/teleport/blob/3b2da5f2573177425fba412c78e23a781badc884/lib/services/role.go#L1606-L1609

Reply:
Done in https://github.com/gravitational/teleport/pull/54960/commits/c022ffeeb39f44f58e49d3b0e35fb90fc94d216d.

Removed v2 suffix and added "legacy" suffix to the v1 checker.